services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParser.java

```diff
@@ -177,6 +179,23 @@ public VenicePath parseResourceUri(String uri, HTTP_REQUEST request) throws Rout
               routerConfig.getSmartLongTailRetryAbortThresholdMs(),
               routerConfig.isComputeFastAvroEnabled(),
               routerConfig.getLongTailRetryMaxRouteForMultiKeyReq());
+          String supportedComputationStrategy =
+              request.headers().get(HttpConstants.VENICE_SUPPORTED_COMPUTATION_STRATEGY);
+          ComputationStrategy computationStrategy =
+              !storeRepository.isReadComputationEnabled(storeName) && supportedComputationStrategy != null
+                  ? ComputationStrategy.valueOf(Integer.parseInt(supportedComputationStrategy))
+                  : ComputationStrategy.SERVER_SIDE_COMPUTATION;
```

Comment:
So if (!storeRepository.isReadComputationEnabled(storeName) && supportedComputationStrategy == null), shall we throw exception here?

I feel the logic here is a little confusing..
I assume the following should be right:
1. If store supports read compute, always use server-side compute.
2. If store doesn't support read compute, and if client-side compute is available, use multiget (client-side compute).
3. if store doesn't support read compute and client-side compute is not available, throw exception.

Reply:
We're on the same page in terms of expected behavior. In case `!storeRepository.isReadComputationEnabled(storeName) && supportedComputationStrategy == null` this block won't have any effect and the exception will be thrown from the original check in `VeniceDispatcher::dispatch`. I suspect it's possible to move that check here, but need to see first how it will affect metric tracking, etc.